api/model/src/main/java/org/projectnessie/model/Reference.java

```diff
@@ -109,6 +117,26 @@ static String toPathString(String name, String hash) {
   @Schema(enumeration = {"branch", "tag"}) // Required to have lower-case values in OpenAPI
   enum ReferenceType {
     BRANCH,
-    TAG
+    TAG;
+
+    public static ReferenceType parse(String referenceType) {
+      try {
+        if (referenceType != null) {
+          return ReferenceType.valueOf(referenceType.toUpperCase(Locale.ROOT));
+        }
+        return null;
+      } catch (IllegalArgumentException e) {
+        return BRANCH;
```

Comment:
I'd default to `TAG` because it is the simplest form of a named reference. A `BRANCH`, for example, can be used in commits/merges. Some new ref type (if it ever appears) will probably have some new use cases too.